crates/node/src/rpc/mod.rs

```diff
@@ -325,22 +351,26 @@ impl<N: FullNodeTypes<Types = TempoNode>> EthTransactions for TempoEthApi<N> {
         &self,
         tx: WithEncoded<Recovered<PoolPooledTx<Self::Pool>>>,
     ) -> impl Future<Output = Result<B256, Self::Error>> + Send {
-        if tx.value().inner().subblock_proposer().is_some() {
-            // Send subblock transactions to the subblocks service.
-            Either::Left(async move {
-                let tx_hash = *tx.value().tx_hash();
-
-                self.subblock_transactions_tx
-                    .send(tx.into_value())
-                    .map_err(|_| {
+        match tx.value().inner().subblock_proposer() {
+            Some(proposer) if self.matches_validator_key(&proposer) => {
+                let subblock_tx = self.subblock_transactions_tx.clone();
+                Either::Left(Either::Left(async move {
+                    let tx_hash = *tx.value().tx_hash();
+
+                    subblock_tx.send(tx.into_value()).map_err(|_| {
                         EthApiError::from(RethError::msg("subblocks service channel closed"))
                     })?;
 
-                Ok(tx_hash)
-            })
-        } else {
-            // Send regular transactions to the transaction pool.
-            Either::Right(self.inner.send_transaction(tx).map_err(Into::into))
+                    Ok(tx_hash)
+                }))
+            }
+            Some(_) => Either::Left(Either::Right(async move {
+                Err(EthApiError::from(RethError::msg(
+                    "subblock transaction rejected: target validator mismatch",
+                ))
+                .into())
+            })),
```

Comment:
I believe we can use `future::err` here instead of an async block.

```rust
Some(_) => Either::Left(Either::Right(future::err(
    EthApiError::from(RethError::msg(
        "subblock transaction rejected: target validator mismatch",
    ))
    .into(),
))),
```